### PR TITLE
Add type info equality function, tests

### PR DESCRIPTION
**Pull request by alnoki**
@movekevin @davidiw @wrwg

The proposed changes add an `aptos_framework::type_info::are_same_type_info` function that returns `true` when both arguments refer to identical `TypeInfo` fields. Tests included.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2268)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/type_info.move [hunk 1]
```diff
@@ -19,11 +19,44 @@ module aptos_framework::type_info {
 
     public native fun type_of<T>(): TypeInfo;
 
+    /// Return `true` if `type_info_1` and `type_info_2` are the same,
+    /// else `false`
+    public fun are_same_type_info(
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/type_info.move:24]: nit: How about name this are_equal?

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/type_info.move (hunk 1 above)
Comment (alnoki) [reply aptos-move/framework/aptos-framework/sources/type_info.move]: Renamed per [`c7813d9`](https://github.com/aptos-labs/aptos-core/pull/2268/commits/c7813d952bf84eb5cdf512e700952de62e024138)

**Review by davidiw — CHANGES_REQUESTED** [review]
Move has some ambiguity, but all structs support `==` and `!=`. Have you tried that?

**Comment by alnoki** [comment]
> Move has some ambiguity, but all structs support `==` and `!=`. Have you tried that?

Thanks for the tip! We [just updated our code](https://github.com/econia-labs/econia/commit/8e6fafa866faac94c15c20b8cf2493253940c086#diff-28b4c952cf1d74c3bdf65d6593aa585066c880531573c82575fb58616e729b87) with the comparator overload and were able to get rid of an entire `util.move` module.

Seeing as this was overlooked by everyone on the issue except you, would it be appropriate to leave the proposed function in there, but private, and explain how it is not needed because of the comparator?

**Comment by wrwg** [comment]
> > Move has some ambiguity, but all structs support `==` and `!=`. Have you tried that?
> 
> Thanks for the tip! We [just updated our code](https://github.com/econia-labs/econia/commit/8e6fafa866faac94c15c20b8cf2493253940c086#diff-28b4c952cf1d74c3bdf65d6593aa585066c880531573c82575fb58616e729b87) with the comparator overload and were able to get rid of an entire `util.move` module.
> 
> Seeing as this was overlooked by everyone on the issue except you, would it be appropriate to leave the proposed function in there, but private, and explain how it is not needed because of the comparator?

Sorry for jumping in late here, but please let's not leave this in there. It is not needed in Move.

The PR now also has some commits which don't belong there. Perhaps close it? The outcome is positive, one function less needed!

**Pull request closed without merging**